tests/test_cli/test_fetch.py

```diff
@@ -304,6 +306,102 @@ class TestFetchAgentRemoteModeError(BaseTestFetchAgentError):
     MODE = "--remote"
 
 
+class TestFetchIPFS(BaseAEATestCase):
+    """Test fetching package from the IPFS registry."""
+
+    dummy_id = (
+        "valory/dummy_agent:bafybeic6plxxhcbokxlce3grbgsm247rgvcb5arwwslm5yv6pt46byr22a"
+    )
+
+    def test_run(self) -> None:
+        """Test run."""
+
+        with mock.patch(
+            "aea.cli.fetch.get_default_remote_registry", return_value=REMOTE_IPFS
+        ), mock.patch.object(
+            IPFSTool,
+            "download",
+            return_value=self.packages_dir_path
+            / "open_aea"
+            / "agents"
+            / "my_first_aea",
+        ), mock.patch(
+            "aea.cli.fetch._fetch_agent_deps"
+        ) as dep_patch:
+            result = self.run_cli_command(
+                "fetch",
+                self.dummy_id,
+                "--remote",
+            )
+
+            assert result.exit_code == 0
+            dep_patch.assert_called_once()
+
+    def test_download_failure(self) -> None:
+        """Test run."""
+
+        with mock.patch(
+            "aea.cli.fetch.get_default_remote_registry", return_value=REMOTE_IPFS
+        ), mock.patch.object(IPFSTool, "download", side_effect=DownloadError):
+
+            with pytest.raises(
+                click.ClickException,
+                match="Error occured while downloading agent",
+            ):
+                self.run_cli_command(
+                    "fetch",
+                    self.dummy_id,
+                    "--remote",
+                )
+
+    def test_os_error(self) -> None:
+        """Test run."""
+
+        with mock.patch(
+            "aea.cli.fetch.get_default_remote_registry", return_value=REMOTE_IPFS
+        ), mock.patch.object(IPFSTool, "download", side_effect=shutil.Error):
+
+            with pytest.raises(
+                click.ClickException,
+            ):
```

Comment:
can you add a `match`?

Reply:
Can't because I'm raising an empty error to see if we handle the os related errors or not

https://github.com/valory-xyz/open-aea/blob/80ff13f6a50162642c7668393d5dc8b278f12612/tests/test_cli/test_fetch.py#L362